cpp/src/benchmark_core.h

```diff
@@ -285,7 +285,37 @@ void bulk_evict_benchmark(Aggregate agg, Experiment exp) {
         std::cerr << force_side_effect << std::endl;
     }
     else {
-        throw std::runtime_error("bulk_evict_benchmark latency not implemented");
+        for (i = exp.iterations - exp.ooo_distance; i < exp.iterations; ++i) {
+            agg.insert(i, 1 + (i % 101));
+        }
+        for (i = 0; i < exp.window_size - exp.ooo_distance; ++i) {
+            agg.insert(i, 1 + (i % 101));
+        }
+
+        if (agg.size() != exp.window_size) {
+            std::cerr << "window is not exactly full; should be " << exp.window_size << ", but is " << agg.size();
+            exit(2);
+        }
+
+        uint64_t start_pos = exp.window_size - exp.ooo_distance;
+        uint64_t stop_pos = exp.iterations - exp.ooo_distance;
+        for (i = start_pos; i < stop_pos; /* nop */) {
+            std::atomic_thread_fence(std::memory_order_acquire);
+            auto before = rdtsc();
+            std::atomic_thread_fence(std::memory_order_release);
+
+            agg.bulkEvict(i - exp.window_size + exp.bulk_size - 1);
```

Comment:
I'm confused ... I would have expected that we would record how long the eviction took before doing the insertions?

Reply:
We could do that, but then what should we measure for the bulk evict & insert benchmark? Like in this case, the insertion time (native bulk or iteration) will dominate the timing.  The current setup doesn't allow us to record/report two separate latency numbers, though it seems like we wanna know:
1. One-round latency (what we're currently measuring)
2. Bulk evict latency (we could obtain this by timing just the `bulkEvict` portion)
3. Bulk insert vs  iteration-based insert latency

Reply:
Yeah, I guess we could do three runs, instead of trying to collect multiple of these from the same run?

Reply:
I'm going try the following if you both think it makes sense. For each iteration, we will just append two numbers (evict latency and insertion latency) to the same vector, so then the output file will have `2N` numbers instead of `N` that it has currently. The burden will be on the processing script: every even-indexed number is an eviction latency and every odd-indexed number is an insertion latency. This is making an assumption that one-round latency (not directly measured) is equal to the sum of these two latency numbers, which should be the case because there is nothing else going there aside from the timing logic.

Reply:
It might be cleaner if we made `latencies` a vector-of-a-vector. That will require changing a bunch of code through the benchmark runners, though.

Reply:
Let's keep the `latencies` field as the overall latency numbers. I'm adding a new vector of vectors of specific latencies. This will be called `extra_latencies`.